### ISSUE TITLE
Client API functions still using hard coded api prefix

### ISSUE THREAD

**Issue by rmainseas** [seas-computing/course-planner#359]
### Description

Client API calls have hard coded base URLs in the prefix. This means that while it may currently work (since our API lives at `/api`and the `process.env..SERVER_URL` environment variable is just set to `/` ) it won't necessarily work for other configurations. 

API functions are currently making requests to `/api/<endpoint`, but prefixed with the value of `process.env.SERVER_URL`. As well as being somewhat confusing to have requests going to `/api/api` - for example:

![image](https://user-images.githubusercontent.com/50675045/123989883-4e302800-d997-11eb-8bdd-26a6b7cfcd1e.png)

...it also has the potential to cause unexpected behavior at deployment if we (for example) decide to change the `/api` prefix

In summary, we should probably remove this hard-coded `/api` prefix from our API calls and provide that value via environment variable instead.

**Comment by jonseitz** [comment]
@rmainseas I think the `/api/api` might be specific to your setup. The value of `process.env.SERVER_URL` is used to prefix our client-side API requests, but also in the call to `setGlobalPrefix` on the server side. So in my local `.env` file, I have `http:// localhost:3001/course-planner`, and when we launch on AWS we'll have that set to `https://computingapps.seas.harvard.edu/course-planner`. 

![2021-07-01_161449](https://user-images.githubusercontent.com/6124793/124184396-81ea7b00-da87-11eb-9a7e-298278374e71.png)


I don't think adding `/api` to the `SERVER_URL` is the right approach, because not all of our server routes are prefixed with `/api` (the health check and authentication routes don't have it). We could have a separate `API_URL_PREFIX` environment variable, and add that to all of our api routes on the client and server side, but personally I'd prefer to have that be a convention we follow than a variable we set.
